packages/memory/src/tools/working-memory.ts

```diff
@@ -1,19 +1,29 @@
 import type { MemoryConfig } from '@mastra/core';
 import { createTool } from '@mastra/core';
 import { z } from 'zod';
+import type { ZodObject } from 'zod';
 
 export const updateWorkingMemoryTool = (memoryConfig?: MemoryConfig) => {
-  return createTool({
-    id: 'update-working-memory',
-    description:
-      'Update the working memory with new information. Always pass data as string to the memory field. Never pass an object.',
-    inputSchema: z.object({
+  const hasSchema = !!memoryConfig?.workingMemory?.schema;
+
+  let inputSchema;
+
+  if (hasSchema) {
+    inputSchema = memoryConfig.workingMemory?.schema as ZodObject<any, any, any>;
+  } else {
+    inputSchema = z.object({
       memory: z
         .string()
         .describe(
-          `The ${!!memoryConfig?.workingMemory?.schema ? 'JSON' : 'Markdown'} formatted working memory content to store. This MUST be a string. Never pass an object.`,
+          `The ${hasSchema ? 'JSON' : 'Markdown'} formatted working memory content to store. This MUST be a string. Never pass an object.`,
```

Comment:
this will always be markdown. Maybe this makes more sense like this:

```suggestion
 const  inputSchema = z.object({
memory: hasSchema ? (memoryConfig.workingMemory?.schema as ZodObject<any, any, any>).describe(
          `The JSON formatted working memory content to store.` :  z 
.string()
.describe(
          `The Markdown formatted working memory content to store. This MUST be a string. Never pass an object.`,
```